views/index.py

```diff
@@ -143,32 +143,60 @@ def registration():
 
 @library.route('/search', methods=['GET'])
 def search():
+    try:
+        user = User.query.get(session['id'])
+        admin = user.has_role('ADMIN')
+    except KeyError:
+        abort(401)
+    except Exception:
+        abort(500)
+
     if request.method == 'GET':
-        try:
-            retrieve_book_data = db.session.query(Book).all()
-        except TimeoutError:
-            return abort(500)
-
-        books = []
-        for row in retrieve_book_data:
-            book = []
-            book.append(row.id)
-            book.append(row.original_title)
-            book.append([])
-            while row.authors is not None:
-                try:
-                    book[2].append(str(row.authors.pop()))
-                except IndexError:
-                    break
-
-            books.append(book)
-        books.sort(key=lambda x: x[2])
-
-        return render_template('search.html',
-                               title='Search',
-                               books=books)
-    else:
-        return abort(405)
+        if not request.args or not request.args.get('query'):
+            form = SearchForm()
+            page = request.args.get('page', 1, type=int)
+            paginate_query = LibraryItem.query.order_by(
+                LibraryItem.title.asc()).paginate(page, 10, False)
+
+            next_url = (url_for('library.search',
+                                page=paginate_query.next_num)
+                        if paginate_query.has_next else None)
+            prev_url = (url_for('library.search',
+                                page=paginate_query.prev_num)
+                        if paginate_query.has_prev else None)
```

Comment:
Try to move next_url and prev_url logic to jinja template. Maybe it will be better, check it :)